src/js/actions/tools.js

```diff
@@ -793,6 +790,12 @@ define(function (require, exports) {
             toolStore = this.flux.store("tool"),
             tools = toolStore.getAllTools();
 
+        var _documentChangeHandler = function () {
+            this.transfer(resetBorderPolicies);
```

Comment:
Also, we can do a lot better here. In particular, the handler should be debounced by a small amount (100 or 200ms), and ideally would only be triggered if the selected layers in the active document changes. Currently, I do not think that this will trigger if you simply switch documents because the document store doesn't know anything about which document is currently active.